src/ground-truth.rkt

```diff
@@ -1,93 +1,86 @@
 #lang racket
 
-(require math/bigfloat
-         rival)
+(require rival)
 
-(require "syntax/types.rkt" "accelerator.rkt"
-         "common.rkt"
-         "compiler.rkt" "timeline.rkt" "config.rkt")
+(require "correct-round.rkt")
 
-(provide eval-progs-real
-         ground-truth-require-convergence 
-         ival-eval
-         make-search-func)
+(provide rival-compile rival-apply rival-analyze
+         (struct-out exn:rival)
+         (struct-out exn:rival:invalid)
+         (struct-out exn:rival:unsamplable)
+         rival-profile-iterations-taken
+         (struct-out discretization))
+(provide (struct-out rival-machine)) ; This is temporary and we want to get rid of it
 
 (define ground-truth-require-convergence (make-parameter #t))
 
-(define (is-samplable-interval repr interval)
-  (define <-bf (representation-bf->repr repr))
+(define (is-samplable-interval disc interval)
+  (define convert (discretization-convert disc))
+  (define distance (discretization-distance disc))
   (define (close-enough? lo hi)
-    (let ([lo* (<-bf lo)] [hi* (<-bf hi)])
-      (or (equal? lo* hi*) (and (number? lo*) (= lo* hi*)))))
+    (= (distance (convert lo) (convert hi)) 0))
```

Comment:
I like the idea of the faithful rounding that we introduce in this PR as well. I will keep in mind that here it can be implemented with some kind of a flag